### DISCUSSION TITLE
Save to_netcdf hangs after using open_mfdataset

### DISCUSSION THREAD

**Discussion by emmaworthington** [pydata/xarray#5801] [Q&A]
Hi,
I'm using open_mfdataset to open a number of NEMO/ORCA12 files for a small region. I regrid the data using xesmf, then I want to save the regridded data using to_netcdf. However, the code just hangs while saving, and appears to be stuck in threading.wait().

I've tried adding lock=False to the open_mfdataset parameters, but it still hangs.

Any ideas? Otherwise I'll load the data separately, do the regridding and merge the regridded data manually.

Thanks

**Comment by mathause** [comment]
There are some open issues on this topic: https://github.com/pydata/xarray/issues?q=is%3Aissue+is%3Aopen+hangs

I never really figured it out and ended up throwing in a `ds = ds.load()` not nice but works (if you have enough memory).
